src/Commands/Files/ConvertFileToPdf.cs

```diff
@@ -0,0 +1,181 @@
+﻿using Microsoft.SharePoint.Client;
+using PnP.Framework.Utilities;
+using System.Management.Automation;
+using PnP.PowerShell.Commands.Base;
+using PnP.PowerShell.Commands.Utilities.REST;
+using System.IO;
+using PnP.PowerShell.Commands.Base.PipeBinds;
+
+namespace PnP.PowerShell.Commands.Files
+{
+    [Cmdlet(VerbsData.Convert, "PnPFileToPdf")]
+    public class ConvertFileToPdf : PnPWebCmdlet
+    {
+        private const string URLTOPATH = "Save to local path";
+        private const string URLASMEMORYSTREAM = "Return as memorystream";
+        private const string UPLOADTOSHAREPOINT = "Upload to SharePoint";
+
+        [Parameter(Mandatory = true, ParameterSetName = URLTOPATH, Position = 0, ValueFromPipeline = true)]
+        [Parameter(Mandatory = true, ParameterSetName = URLASMEMORYSTREAM, Position = 0, ValueFromPipeline = true)]
+        [Parameter(Mandatory = true, ParameterSetName = UPLOADTOSHAREPOINT, Position = 0, ValueFromPipeline = true)]
+        [Alias("ServerRelativeUrl", "SiteRelativeUrl")]
+        public string Url;
+
+        [Parameter(Mandatory = true, ParameterSetName = URLTOPATH)]
+        public string Path = string.Empty;
+
+        [Parameter(Mandatory = false, ParameterSetName = URLTOPATH)]
+        public SwitchParameter Force;
+
+        [Parameter(Mandatory = true, ParameterSetName = UPLOADTOSHAREPOINT)]
+        [ValidateNotNullOrEmpty]
+        public FolderPipeBind DocumentLibrary;
+
+        [Parameter(Mandatory = false, ParameterSetName = UPLOADTOSHAREPOINT)]
+        public SwitchParameter Checkout;
+
+        [Parameter(Mandatory = false, ParameterSetName = UPLOADTOSHAREPOINT)]
+        public string CheckInComment = string.Empty;
+
+        [Parameter(Mandatory = false, ParameterSetName = UPLOADTOSHAREPOINT)]
+        public CheckinType CheckinType = CheckinType.MinorCheckIn;
+
+        [Parameter(Mandatory = false, ParameterSetName = UPLOADTOSHAREPOINT)]
+        public SwitchParameter Approve;
+
+        [Parameter(Mandatory = false, ParameterSetName = UPLOADTOSHAREPOINT)]
+        public string ApproveComment = string.Empty;
+
+        [Parameter(Mandatory = false, ParameterSetName = UPLOADTOSHAREPOINT)]
+        public SwitchParameter Publish;
+
+        [Parameter(Mandatory = false)]
+        public string PublishComment = string.Empty;
```

Comment:
We dont need these parameters. 
Lets keep it simple.

Download file locally
Upload it to SharePoint folder
Make it available as memory stream

Reply:
Changes done in the latest commit